browser/brave_content_browser_client.cc

```diff
@@ -544,6 +548,69 @@ BraveContentBrowserClient::AllowWebBluetooth(
   return ContentBrowserClient::AllowWebBluetoothResult::BLOCK_GLOBALLY_DISABLED;
 }
 
+bool BraveContentBrowserClient::CanCreateWindow(
+    content::RenderFrameHost* opener,
+    const GURL& opener_url,
+    const GURL& opener_top_level_frame_url,
+    const url::Origin& source_origin,
+    content::mojom::WindowContainerType container_type,
+    const GURL& target_url,
+    const content::Referrer& referrer,
+    const std::string& frame_name,
+    WindowOpenDisposition disposition,
+    const blink::mojom::WindowFeatures& features,
+    bool user_gesture,
+    bool opener_suppressed,
+    bool* no_javascript_access) {
+  content::WebContents* contents =
+      content::WebContents::FromRenderFrameHost(opener);
+
+  if (google_sign_in::IsGoogleSignInFeatureEnabled() &&
+      google_sign_in::IsGoogleAuthRelatedRequest(target_url, opener_url)) {
+    PrefService* prefs =
+        static_cast<Profile*>(contents->GetBrowserContext())->GetPrefs();
+    if (!google_sign_in::IsGoogleSignInPrefEnabled(prefs)) {
+      return false;
+    }
+    // check Google Sign-In permission
+    content::PermissionControllerDelegate* permission_controller =
+        contents->GetBrowserContext()->GetPermissionControllerDelegate();
+    auto status = google_sign_in::GetCurrentGoogleSignInPermissionStatus(
+        permission_controller, contents, opener_url);
+
+    if (status == blink::mojom::PermissionStatus::GRANTED) {
+      // return ChromeContentBrowserClient impl, proceed normally
+      return ChromeContentBrowserClient::CanCreateWindow(
```

Comment:
I think this should be called first because it handles things like generic popup blocking